pagefaultgames/pokerogue
# [Ability] Implement Ice Face

## What are the changes?
- implement ice face
- **making this bold because it's easy to miss on files changed: renamed audio file to match Eiscue's no-ice form key**

## Why am I doing these changes?
- implement ice face

## What did change?
- new ability attr `IceFaceImmunityAbAttr`
- new BattlerTag `IceFaceTag`
- renamed Eiscue's no-ice audio file

### Screenshots/Videos
- Ice face is hit with physical move

https://github.com/pagefaultgames/pokerogue/assets/68144167/3623b71b-7e57-48b4-8c11-5d7b58c659c4



- Ice face is hit with special move

https://github.com/pagefaultgames/pokerogue/assets/68144167/97b5a7de-5088-4302-a0ad-98c3eb577b79



- No ice is hit with physical

https://github.com/pagefaultgames/pokerogue/assets/68144167/958002b5-5c84-459d-a194-71f8d9a7f634



- No ice used hail- reverts to ice face. if noice is on the field with already active hail/snow, does not revert to ice face

https://github.com/pagefaultgames/pokerogue/assets/68144167/096fc235-9ad0-4b49-ad3a-c3793ccc62e5




- No ice summoned under Hail weather

https://github.com/pagefaultgames/pokerogue/assets/68144167/678cd96f-a5f9-4ed4-b9c5-610da8d15510



- No ice will revert to ice face on _pokerogue's context of battle end_, regardless of weather


https://github.com/pagefaultgames/pokerogue/assets/68144167/e19309f8-7daf-4caf-9d38-2c09b4ccf372






https://github.com/pagefaultgames/pokerogue/assets/68144167/9497e0c2-6135-4c86-92dc-a458d6304f8d





## How to test the changes?
[ability description](https://bulbapedia.bulbagarden.net/wiki/Ice_Face_(Ability))
- in `overrides.ts`:
```
export const STARTING_BIOME_OVERRIDE: Biome = Biome.SNOWY_FOREST;
export const STARTER_SPECIES_OVERRIDE: Species | integer = Species.EISCUE; 
export const MOVESET_OVERRIDE: Array<Moves> = [Moves.HAIL];
// set physical and special moves to check for the form change and immunity to physical
export const OPP_MOVESET_OVERRIDE: Array<Moves> = [Moves.DRAGON_PULSE, Moves.DRAGON_PULSE, Moves.TACKLE, Moves.TACKLE];
```

## Checklist
- ~[] Are the changes visual?~
